GDJS/scripts/generate-esm-types.js

```diff
@@ -0,0 +1,30 @@
+const {
+  compileESModuleTypeDefinitions,
+} = require('./lib/build-esm-library-types');
+const { getAllInOutFilePaths } = require('./lib/runtime-files-list');
+
+(async () => {
+  // Generate the output file paths
+  const { allESMInOutFilePaths } = await getAllInOutFilePaths({
+    bundledOutPath: __dirname, // Dummy directory name, since we do not care about compilation destination.
+  });
+
+  const entryPointsForNamespace = new Map();
+  for (const { inPath, namespace } of allESMInOutFilePaths) {
+    const entryPoints = entryPointsForNamespace.get(namespace) || [];
+    entryPoints.push(inPath);
+    entryPointsForNamespace.set(namespace, entryPoints);
```

Comment:
The `set` can be called only when the array is created. It will make it clearer that `inPath` are group by `namespace`.

Reply:
I am not sure what you mean, could you make a change suggestion? (Ctrl-g in the GitHub comment text box)

Reply:
Something like:
```suggestion
    let entryPoints = entryPointsForNamespace.get(namespace);
    if (!entryPoints) {
      entryPoints = [];
      entryPointsForNamespace.set(namespace, entryPoints);
    }
    entryPoints.push(inPath);
```

Reply:
Ah, I see. I'll do that!